hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -612,6 +613,29 @@ public List<HddsProtos.Node> queryNode(
     return result;
   }
 
+  @Override
+  public HddsProtos.Node queryNode(UUID uuid)
+      throws IOException {
+    HddsProtos.Node result = null;
+    for (DatanodeDetails node : scm.getScmNodeManager().getAllNodes()) {
+      try {
+        if (node.getUuid().equals(uuid)) {
+          NodeStatus ns = scm.getScmNodeManager().getNodeStatus(node);
```

Comment:
Node manager has an API to get a node by UUID directly:

```
  public DatanodeDetails getNodeByUuid(UUID uuid);

  public DatanodeDetails getNodeByUuid(String uuid);
```

So it would be better to use that than iterate all nodes.

Reply:
Thank you for the review! I have made the changes required to incorporate your suggestion and resolved the conflicts as well. Could you please review it and approve the workflows if the patch is good to go?